### DISCUSSION TITLE
How to use \includegraphics on this?

### DISCUSSION THREAD

**Discussion by gfbdy** [ismaelw/laratex#12] [Q&A] [answered]
I tried tinkering the config file adding ` -interaction=nonstopmode -enable-write18` but it just doesn't work.
On whether http and local image, I have no clue.

`\write18{wget http://www.some-site.com/path/to/image.png}
\includegraphics{image.png}`

Hope somebody can help be 'coz this package is so so good! If images show, I'm never going back to domPDF and Chrome Print!

**Comment by ismaelw** [comment, marked as answer]
Hi there!

Thank you so much for your kind words.

In my readme there is a section that talks about graphics ([here](https://github.com/ismaelw/laratex#using-graphics-inside-of-your-latex-files)).

As to my understanding the latex compiler looks for the graphics inside the path it got started in. Where this is in the context of running it from a web application I still couldn't figure out.

But how I am able to get it to work is using the absolute path to my file. What I am doing im my usecases is to define a PHP variable in my latex blade file for the base path to all my graphics so I don't have to rewrite this path all of the time.

Please tell me if this answers your question. If not - I am here to help.

Kind regards
Ismael

**Reply by ismaelw** [reply]
@gfbdy Was my answer a help for your question?

**Reply by ismaelw** [reply]
@gfbdy Feel free to reply to my answer. Otherwise I will close this discussion soon due to inactivity :)

**Reply by gfbdy** [reply]
@ismaelw Answered my problem. Thank you so much! Does this package have a function to strip LaTeX-interrupting symbols like @ _ {} [] $?

**Reply by ismaelw** [reply]
Hi @gfbdy 
Thank you so much for your feedback.
As to your other question, I didn't implement any content check to prevent LaTeX exceptions like this directly.

Until now I always used overleaf to check my documents for the correct syntax, and then replaced special characters while passing the variables to my LaTeX blade file.

Does that somehow make sense?

If you have any ideas on how to globally fix issued like this, feel free to contribute or to write down your ideas.

**Reply by ismaelw** [reply]
hi @gfbdy 

In the latest version we implemented a latex directive which strips LaTeX-sensitive characters. You can use it using @latex($variable) in your blade file. This should remove all characters like {} [] $ etc. Feel free to test it out :)
